ibm/resource_ibm_private_dns_permitted_network.go

```diff
@@ -0,0 +1,154 @@
+		Delete:   resourceIBMPrivateDnsPermittedNetworkDelete,
+		Exists:   resourceIBMPrivateDnsPermittedNetworkExists,
+		Importer: &schema.ResourceImporter{},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		CustomizeDiff: customdiff.Sequence(
+			func(diff *schema.ResourceDiff, v interface{}) error {
+				return resourceTagsCustomizeDiff(diff)
+			},
+		),
+
+		Schema: map[string]*schema.Schema{
+			pdnsInstanceID: {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false,
+			},
+
+			pdnsZoneID: {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false,
+			},
+
+			pdnsVpcCRN: {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false,
+			},
+		},
+	}
+}
+func resourceIBMPrivateDnsPermittedNetworkCreate(d *schema.ResourceData, meta interface{}) error {
+	sess, err := meta.(ClientSession).PrivateDnsClientSession()
+	if err != nil {
+		return err
+	}
+
+	instanceID := d.Get(pdnsInstanceID).(string)
+	zoneID := d.Get(pdnsZoneID).(string)
+	vpcCRN := d.Get(pdnsVpcCRN).(string)
+
+	createPermittedNetworkOptions := sess.NewCreatePermittedNetworkOptions(instanceID, zoneID)
+	permittedNetworkCrn, err := sess.NewPermittedNetworkVpc(vpcCRN)
+
+	createPermittedNetworkOptions.SetPermittedNetwork(permittedNetworkCrn)
+	createPermittedNetworkOptions.SetType(dnssvcsv1.CreatePermittedNetworkOptions_Type_Vpc)
+	response, _, err := sess.CreatePermittedNetwork(createPermittedNetworkOptions)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s/%s", instanceID, zoneID, *response.ID))
+	if err != nil {
+		log.Printf("[DEBUG]  err %s", isErrorToString(err))
+		return err
+	}
+
+	return resourceIBMPrivateDnsPermittedNetworkRead(d, meta)
+}
+
+func resourceIBMPrivateDnsPermittedNetworkRead(d *schema.ResourceData, meta interface{}) error {
+
+	sess, err := meta.(ClientSession).PrivateDnsClientSession()
+	if err != nil {
+		return err
+	}
+
+	id_set := strings.Split(d.Id(), "/")
+	getPermittedNetworkOptions := sess.NewGetPermittedNetworkOptions(id_set[0], id_set[1], id_set[2])
+	_, _, err = sess.GetPermittedNetwork(getPermittedNetworkOptions)
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceIBMPrivateDnsPermittedNetworkUpdate(d *schema.ResourceData, meta interface{}) error {
+	return resourceIBMPrivateDnsPermittedNetworkRead(d, meta)
+}
+
+func resourceIBMPrivateDnsPermittedNetworkDelete(d *schema.ResourceData, meta interface{}) error {
+	sess, err := meta.(ClientSession).PrivateDnsClientSession()
+	if err != nil {
+		return err
+	}
+
+	id_set := strings.Split(d.Id(), "/")
+	deletePermittedNetworkOptions := sess.NewDeletePermittedNetworkOptions(id_set[0], id_set[1], id_set[2])
+	_, _, reqErr := sess.DeletePermittedNetwork(deletePermittedNetworkOptions)
+
+	if reqErr != nil {
```

Comment:
check if reqErr !=  nil and status code is not 404 then only return err
As per new Terraform v0.12 if the resource is deleted outside terraform using cli or UI we should not return error
https://www.terraform.io/docs/extend/writing-custom-providers.html#implementing-destroy